src/components/decimal/decimal.test.tsx

```diff
@@ -0,0 +1,1406 @@
+    [1, 8, "1", "1"],
+    [2, 3, "1,34.56", "134.56"],
+    [2, 4, "1,4.56", "14.56"],
+    [2, 5, "1,.56", "1,.56"],
+    [2, 6, "1,56", "156"],
+    [2, 7, "1,6", "16"],
+    [2, 8, "1,", "1,"],
+    [3, 4, "1,24.56", "124.56"],
+    [3, 5, "1,2.56", "12.56"],
+    [3, 6, "1,256", "1256"],
+    [3, 7, "1,26", "126"],
+    [3, 8, "1,2", "12"],
+    [4, 5, "1,23.56", "123.56"],
+    [4, 6, "1,2356", "12356"],
+    [4, 7, "1,236", "1236"],
+    [4, 8, "1,23", "123"],
+    [5, 6, "1,23456", "123456"],
+    [5, 7, "1,2346", "12346"],
+    [5, 8, "1,234", "1234"],
+    [6, 7, "1,234.6", "1234.6"],
+    [6, 8, "1,234.", "1,234."],
+    [7, 8, "1,234.5", "1234.5"],
+  ])(
+    "allows deletion of content using the delete key for a selection starting at index %s and ending at index %s",
+    async (selectionStart, selectionEnd, rawValue, sanitisedValue) => {
+      const user = userEvent.setup();
+      render(<Decimal defaultValue="1234.56" />);
+
+      const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+      decimalInput.focus();
+      decimalInput.setSelectionRange(selectionStart, selectionEnd);
+      await user.keyboard("{Delete}");
+
+      expect(decimalInput).toHaveValue(rawValue);
+      expect(screen.getByTestId("hidden-input")).toHaveValue(sanitisedValue);
+    }
+  );
+
+  // as above, a simple cursor placement (no selection) at the start or end need special handling
+  it("deletes the fisrt character when delete is pressed with the cursor at the start of the input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="1234.56" />);
+
+    const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+    decimalInput.focus();
+    await user.keyboard("{ArrowLeft}");
+    await user.keyboard("{Delete}");
+
+    expect(decimalInput).toHaveValue(",234.56");
+    expect(screen.getByTestId("hidden-input")).toHaveValue(",234.56");
+  });
+
+  it("doesn't delete anything when delete is pressed with the cursor at the end of the input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="1234.56" />);
+
+    const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+    decimalInput.focus();
+    await user.keyboard("{ArrowRight}");
+    await user.keyboard("{Delete}");
+
+    expect(decimalInput).toHaveValue("1,234.56");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("1234.56");
+  });
+
+  it.each([",", ".", "£", "$", "%", "@", "!", "*", "#"])(
+    "should not format multiple `%s` characters",
+    async (char) => {
+      const user = userEvent.setup();
+      render(<Decimal />);
+
+      const input = char.repeat(4);
+      await user.type(screen.getByRole("textbox"), input);
+      await user.tab();
+
+      expect(screen.getByRole("textbox")).toHaveValue(input);
+      expect(screen.getByTestId("hidden-input")).toHaveValue(input);
+    }
+  );
+
+  it("calls the onKeyPress callback when a key is pressed", async () => {
```

Comment:
comment: good to see us using a deprecated keyboard event 🤦, I'll see if we have a ticket to update this and raise one if not

Reply:
FE-6766 raised